AppCenter/AppCenter/MSACLogger.m

```diff
@@ -78,12 +79,12 @@ + (void)logMessage:(MSACLogMessageProvider)messageProvider
               file:(const char *)file
           function:(const char *)function
               line:(uint)line {
-    
-    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void){
-        if (currentLogHandler) {
-            currentLogHandler(messageProvider, loglevel, tag, file, function, line);
-        }
-    });
+
+  dispatch_async([[MSACNSLogQueueManager sharedManager] mySerialQueue], ^{
```

Comment:
Better to put `dispatch_async` under the condition `if (currentLogHandler)` to avoid dispatching if there is no log handler.